feature-libs/customer-ticketing/components/customer-ticketing-create/customer-ticketing-create-dialog/customer-ticketing-create-dialog.component.ts

```diff
@@ -10,7 +13,18 @@ export class CustomerTicketingCreateDialogComponent
   extends CustomerTicketingDialogComponent
   implements OnInit
 {
+  ticketCategories: Array<Category>;
+  ticketAssociatedObjects: Array<AssociatedObjects>;
+
   ngOnInit(): void {
+    this.customerTicketingFacade.getTicketCategories().subscribe((category) => {
```

Comment:
If you were to subscribe in the component .ts file, you would need to always unsubscribe on component destroy as it will cause memory leaks.

You have options:

1 - let angular handle everything by using the `async` pipe in the template ('.html'). What this does for you is it subscribes to the observable, and unsubscribes to it whenever the component gets destroyed.


2 - Do what you're doing here, but you would need to create a subscription object, such as `const subscriptions = new Subscription()` from `rxjs`. and add every subscribed things in the component level. You would need to implement the `OnDestroy` to handle the lifecycle when the component gets destryoed, and therefore unsubscribe in one shot.

Reply:
I would prefer, you would go with the 1st approach as it's less code and more compact. 

That would also mean, you would make the variables of type Observable< insert type here >